app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,288 @@
+        // ATM tacking this on via watching the current project.
+        var watchNotifications = function(projectName, cb) {
+          if(!projectName) {
+            return;
+          }
+          notificationListener = $rootScope.$on('NotificationsService.onNotificationAdded', cb);
+        };
+
+        var deregisterNotificationListener = function() {
+          notificationListener && notificationListener();
+        };
+
+        var unread = function(notifications) {
+          return _.filter(notifications, 'unread');
+        };
+
+        // returns a count for each type of notification, example:
+        // {Normal: 1, Warning: 5}
+        var countUnreadNotificationsForGroup = function(group) {
+          return _.countBy(unread(group.notifications), function(notification) {
+            return notification.status;
+          });
+        };
+
+        // currently we only show 1 at a time anyway
+        var countUnreadNotificationsForAllGroups = function() {
+          _.each(drawer.notificationGroups, function(group) {
+            group.counts = countUnreadNotificationsForGroup(group);
+          });
+        };
+
+        var countAllUnread = function(group) {
+          return _.reduce(group.counts, function(result, count) {
+            return result + count;
+          }, 0);
+        };
+
+        var sortNotifications = function(notifications) {
+          return _.sortBy(notifications, function(notification) {
+            return -(new Date(notification.lastTimestamp));
+          });
+        };
+
+        var sortNotificationGroups = function(groupsMap) {
+          // convert the map into a sorted array
+          var sortedGroups = _.sortBy(groupsMap, function(group) {
+            return group.heading;
+          });
+          // and sort the notifications under each one
+          _.each(sortedGroups, function(group) {
+            group.notifications = sortNotifications(group.notifications);
+            group.counts = countUnreadNotificationsForGroup(group);
+          });
+          return sortedGroups;
+        };
+
+        var formatFilteredEvents = function(eventMap) {
+          var filtered = {};
+          _.each(eventMap, function(event) {
+            if(EventsService.isImportantEvent(event)) {
+              ensureProjectGroupExists(filtered, event.metadata.namespace);
+              filtered[event.metadata.namespace].notifications.push({
+                unread:  !EventsService.isRead(event),
+                message: event.message,
+                lastTimestamp: event.lastTimestamp,
+                metadata: event.metadata,
+                involvedObject: event.involvedObject,
+                name: event.involvedObject.name,
+                kind: event.involvedObject.kind,
+                namespace: event.involvedObject.namespace,
+                status: event.type,
+                reason: event.reason,
+                actions: null
+              });
```

Comment:
I'd rather just pass the event object through instead of trying to copy fields and flatten it.

```js
{
  unread: !EventsService.isRead(event),
  event: event
}
```

I think I'd just leave actions `undefined` instead of explicitly setting to null. Slight preference for making the property `read` instead of `unread` (and flipping the logic that uses it)

```js
{
  read: EventsService.isRead(event),
  event: event
}
```